File: javascript/test/tests/primes.js
'use strict';

(function(){
	QUnit.module('Main');
	
	QUnit.test('isPrime()', function(assert){
		assert.equal(primes.isPrime(0), false, '0 is not prime.');
		assert.equal(primes.isPrime(1), false, '1 is not prime.');
		assert.equal(primes.isPrime(2), false, '2 is prime.');
		assert.equal(primes.isPrime(-15), false, '-15 is not prime.');
		assert.equal(primes.isPrime(7907), true, '7907 is prime.');
		assert.equal(primes.isPrime(3.14), false, '3.14 is not prime.');
		assert.equal(primes.isPrime(5), true, '5 is prime.');
	});
	
	QUnit.test('next()', function(assert){
		assert.equal(primes.next(0), 2, 'Next after 0 is 2.');
		assert.equal(primes.next(2), 3, 'Next after 2 is 3.');
		assert.equal(primes.next(-15), 2, 'Next after -15 is 2.');
		assert.equal(primes.next(509), 521, 'Next after 509 is 521.');
		assert.equal(primes.next(509), 521, 'Next after 6.5 is 7.');
		assert.equal(primes.next(3.14), 5, 'Next after 3.14 is 5.');
		assert.equal(primes.next(7907), 7919, 'Next after 7907 is 7919.');
	});
})();

File: c-plus-plus/Primes/Primes/Primes.cpp
// Primes.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "input.h"
#include "Prime.h"
using namespace std;

/**
Shows the main menu.
@returns {int} The user's menu choice.
*/
int main_menu()
{
	int choice;

	do
	{
		cout << "Please choose an option:\n";
		cout << "1. The next prime number after X\n";
		cout << "0. Exit\n";
		choice = primes::input::get_int(cin);
	} while (choice < 0 || choice > 1);

	return choice;
}

int _tmain(int argc, _TCHAR* argv[])
{
	primes::Prime prime;

	int choice;

	do
	{
		choice = main_menu();
		switch (choice)
		{
		case 1:
			// Next prime after X
			cout << "\nEnter a number for X:\n";
			long x = primes::input::get_long(cin);
			cout << "The next prime after " << x << " is " << prime.next(x) << "\n";
			break;
		}

		cout << "\n";
	} while (choice != 0);

	return 0;
}

